### DISCUSSION TITLE
Restart container systemd unit file generated via Quadlet with dependencies

### DISCUSSION THREAD

**Discussion by giovannibaratta** [containers/podman#23122] [Q&A] [answered]
Hello,
I have two containers with their respective Quadlet (generated using an Ansible module) and systemd services. One container depends on the other. When I try to restart with systemd the container that is a dependency of the other one, the operation generates the following error:

```
Error: container c01d1047396777253a24e20194b64a28eccc5c48577348ea60748b8bad2a93f4 has dependent containers which must be removed before it: a1cdc538c286cd5bf1b8c25c87b0c9eed666dc701a706aa56b6a129253479b87: container already exists
```

Quadlet 1
```ini
[Container]
ContainerName=coder-db
Image=docker.io/postgres:16.3-alpine
Volume=/app/coder/work/database:/var/lib/postgresql/data
...
[Install]
WantedBy=default.target
```

Quadlet 2
```ini
[Container]
ContainerName=coder
Image=ghcr.io/coder/coder:v2.12.2
PodmanArgs=--requires coder-db
...
[Install]
WantedBy=default.target
```

I think the problem is that the systemd of Quadlet 1 does not use the `--depend` flag in the ExecStop command.

Systemd service Quadlet 1
```ini
# /run/systemd/generator/coder-db.service
# Automatically generated by /usr/lib/systemd/system-generators/podman-system-generator
#
[X-Container]
ContainerName=coder-db
...

[Service]
Environment=PODMAN_SYSTEMD_UNIT=%n
KillMode=mixed
ExecStop=/usr/bin/podman rm -v -f -i --cidfile=%t/%N.cid
ExecStopPost=-/usr/bin/podman rm -v -f -i --cidfile=%t/%N.cid
...
```

Is there any equivalent for PodmanArgs that alters the ExecStop command? Should I just remove the dependency of Quadlet 2 and handle it in another way ?

**Comment by rhatdan** [comment]
Would it be better to setup the one  quadlet to depend on the  other using systemd syntax?

**Comment by Luap99** [comment, marked as answer]
Dependencies should be managed via systemd here, not podman.

You can use `Requires=` or `BindsTo=` depeding on what behavior you likein unit 2 with the name of unit 1. 

see https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html#Requires=

In general the systemd dependency management is much better. By using `--requires coder-db` it doesn't actually start unit 1.

**Reply by giovannibaratta** [reply]
Thank you, I will use this approach.
